### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by apozsuse**
This PR includes:
- Improvement for the unit test coverage for lib/ec2uploadimg.py
- Typo fix fin call function
- Fix bug for possible infinitive recursive call in _clean_up() call in case `waiter.wait` gives an exception:
  In the `_clean_up` function, if `repeat_count` < `wait_count` and `waiter.wait` gives an exception, `wait_status` is 1 and in `_check_wait_status` we can call (when `repeat_count` == `wait_count`) `_clean_up` again and we can get in an infinite loop.
We should use the `skip_cleanup` flag set to True when calling `_check_wait_status` from `_clean_up` function

### CODE REVIEW

**Review by smarlowucf — APPROVED** [review]

File: tests/test_libec2uploadimg.py [hunk 1]
```diff
@@ -0,0 +1,2102 @@
+#!/usr/bin/python3
+#
+# Copyright (c) 2015 SUSE Linux GmbH.  All rights reserved.
```
Comment (rjschwei) [inline tests/test_libec2uploadimg.py:3]: Last I checked it was 2023

File: tests/test_libec2uploadimg.py [hunk 2]
```diff
@@ -0,0 +1,2102 @@
...
+# You should have received a copy of the GNU General Public License
+# along with ec2utils. If not, see
+# <http://www.gnu.org/licenses/>.
+#
+
+# import datetime
+# import dateutil.relativedelta
+import logging
+import pytest
+# import time
```
Comment (rjschwei) [inline tests/test_libec2uploadimg.py:26]: No commented imports, if we don't need them please remove.

**Review by rjschwei — CHANGES_REQUESTED** [review]
Couple of minor things. Thanks for the effort

File: tests/test_libec2uploadimg.py (hunk 2 above)
Comment (apozsuse) [reply tests/test_libec2uploadimg.py]: done!

**Pull request merged**